ElectronClient/app.js

```diff
@@ -270,6 +270,20 @@ class Application extends BaseApplication {
 			webFrame.setZoomFactor(Setting.value('windowContentZoomFactor') / 100);
 		}
 
+		if (action.type == 'SETTING_UPDATE_ONE' &&
+			action.key === 'preferredLightTheme' &&
+			!bridge().shouldUseDarkColors() &&
+			Setting.value('themeAutoDetect')) {
+			Setting.setValue('theme', action.value);
+		}
+
+		if (action.type == 'SETTING_UPDATE_ONE' &&
+			action.key === 'preferredDarkTheme' &&
+			bridge().shouldUseDarkColors() &&
+			Setting.value('themeAutoDetect')) {
+			Setting.setValue('theme', action.value);
+    }
```

Comment:
This would not be run when the `themeAutoDetect` checkbox is changed, but it should. For example, consider the example below:

- My system is in dark theme, and I have auto-detect set to "off", with Joplin light theme selected
- I set auto-detect to "on"
- I press OK

Nothing happens, even though the theme should be set to "Dark" like my system.

Also please create a function `detectTheme()` in this file that returns the right theme depending on the values of `shouldUseDarkColors`, `themeAutoDetect` and `preferredLightTheme`/`preferredDarkTheme`. Then use this function in the reducer.

Also make sure that the theme is set correctly when the app starts. If "autoDetect" is on, there should be a call to "detectTheme" and the theme set to the right one.

Reply:
> This would not be run when the themeAutoDetect checkbox is changed, but it should

Ohh I missed this edge case 😕 I'll work on it and test again to make sure it works fine with all edge cases

> Also make sure that the theme is set correctly when the app starts. If "autoDetect" is on, there should be a call to "detectTheme" and the theme set to the right one.

Thank you for your detailed feedback, I'll work on it 👍

Reply:
By the way Electron also emits an event when the system theme has changed. This should be used to update the theme when the app is running. https://www.electronjs.org/docs/api/native-theme